rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2148,23 +2240,28 @@ extern void ppu_precompile(std::vector<std::string>& dir_queue)
 
 	atomic_t<usz> fnext = 0;
 
-	shared_mutex sprx_mtx;
+	shared_mutex sprx_mtx, ovl_mtx;
 
-	named_thread_group workers("SPRX Worker ", utils::get_thread_count(), [&]
+	named_thread_group workers("SPRX Worker ", std::min<u32>(utils::get_thread_count(), ::size32(file_queue)), [&]
 	{
-		for (usz func_i = fnext++; func_i < file_queue.size(); func_i = fnext++)
+		for (usz func_i = fnext++; func_i < file_queue.size(); func_i = fnext++, g_progr_fdone++)
 		{
-			std::string path = std::as_const(file_queue)[func_i].first;
+			if (Emu.IsStopped())
+			{
+				continue;
+			}
 
-			ppu_log.notice("Trying to load SPRX: %s", path);
+			auto [path, type, offset]  = std::as_const(file_queue)[func_i];
 
-			// Load MSELF or SPRX
+			ppu_log.notice("Trying to load: %s", path);
+
+			// Load MSELF, SPRX or SELF
 			fs::file src{path};
 
-			if (u64 off = file_queue[func_i].second)
+			if (u64 off = offset)
 			{
 				// Adjust offset for MSELF
-				src.reset(std::make_unique<file_view>(std::move(src), off));
+				if (src) src.reset(std::make_unique<file_view>(std::move(src), off));
```

Comment:
Check has no effect.

Reply:
It has actually, in case of the filesystem behaves unexpectedly and src is null. (std::make_unique is never returns null fs::file so internal m_file may crash on usage)

Reply:
Okay, if you want to check file for open errors, put a check somewhere directly after its creation.

Reply:
Fixed.